packages/astro/src/vite-plugin-build-css/index.ts

```diff
@@ -56,22 +55,27 @@ export function rollupPluginAstroBuildCSS(options: PluginOptions): VitePlugin {
   const { astroPageStyleMap, astroStyleMap, chunkToReferenceIdMap, pureCSSChunks } = options;
   const styleSourceMap = new Map<string, string>();
 
-  let viteTransform: TransformHook;
-
   return {
     name: PLUGIN_NAME,
 
-    enforce: 'pre',
-
     configResolved(resolvedConfig) {
-      viteTransform = getViteTransform(resolvedConfig);
 
-      const viteCSSPost = resolvedConfig.plugins.find((p) => p.name === 'vite:css-post');
-      if (viteCSSPost) {
+      const plugins = resolvedConfig.plugins as VitePlugin[];
+      const viteCSSPostIndex = resolvedConfig.plugins.findIndex((p) => p.name === 'vite:css-post');
```

Comment:
nit: a comment above all of this explaining why this is needed would be great. Also would help readers know that all we are doing is moving plugins around, and not calling or making changes to any of them.